### PR TITLE
Hide Extract Components, Modal Image Library

### PR DESCRIPTION
**Pull request by AustinShutt**
This modification adds the ability to toggle between hide and extract mode as well as puts the image gallery into a modal window on top of all other elements. Still to be added to the modal window is a button to upload your own image from local files for secret/cover images. As well as upload of image from local system for the extraction process. 

Let me know if you see any issues or have any questions! Thanks!

### CODE REVIEW

**Comment by AustinShutt** [comment]
StegoSource_Update_1

File: Application/client/src/App.js [hunk 1]
```diff
@@ -1,54 +1,28 @@
...
-  const [selectedImage, setSelectedImage] = useState(null);
-
-  // Callback function to be passed to GridGallery
-  const handleImageSelect = (index, image) => {
-    console.log('Selected Image:', image);
-    setSelectedImage(image);
+  //Callback function to decide which Elements are visible either hiding or extraction
+  const handleToggle = () => {
+    setComponentAVisible(!isComponentAVisible);
+    console.log("Hello from toggle in app.js")
```
Comment (TreytonPerkins) [inline Application/client/src/App.js:15]: Do we want to leave this in?

**Review by TreytonPerkins — APPROVED** [review]
Looks good!

File: Application/client/src/App.js (hunk 1 above)
Comment (AustinShutt) [reply Application/client/src/App.js]: Good catch! Thanks

**Pull request merged**